src/textarea/Textarea.tsx

```diff
@@ -0,0 +1,250 @@
+    if (this.props.defaultValue !== defaultValue) {
+      this.setState({
+        value: defaultValue,
+      })
+    }
+    if (focus && this.props.focus !== focus) {
+      this.textarea.focus()
+    }
+  }
+
+  onFocus = (event: React.FocusEvent<HTMLTextAreaElement>) => {
+    this.setState({
+      hasFocus: true,
+    })
+    this.props.onFocus(event)
+  }
+
+  onBlur = (event: React.FocusEvent<HTMLTextAreaElement>) => {
+    if (!event.relatedTarget || event.relatedTarget !== this.buttonRef) {
+      this.setState({
+        hasFocus: false,
+      })
+      this.props.onBlur(event)
+    }
+  }
+
+  onTextAreaChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    this.setState(
+        {
+          value: event.target.value,
+        },
+        this.onChange,
+    )
+  }
+
+  onChange = () => {
+    this.props.onChange({
+      name: this.props.name,
+      value: this.state.value,
+    })
+
+    // Fit to content.
+    if (this.props.fitContent && this.textareaRef && this.textareaRef.current) {
+      this.textareaRef.current.style.height = '0'
+      const verticalPadding = 16
+      this.textareaRef.current.style.height =
+          `${this.textareaRef.current.scrollHeight - 2 * verticalPadding}px`
+
+    }
+  }
+
+  ref = (textarea: HTMLTextAreaElement) => {
+    this.textarea = textarea
+    this.props.fieldRef(textarea)
+  }
+
+  render() {
+    const {
+      className,
+      placeholder,
+      name,
+      id,
+      labelledBy,
+      label,
+      error,
+      disabled,
+      readOnly,
+      onFocus,
+      onBlur,
+      autoFocus,
+      required,
+      maxLength,
+      autoCorrect,
+      autoComplete,
+      title,
+      buttonTitle,
+      focusBorder,
+      buttonIcon,
+      onButtonClick,
+      fitContent,
+    } = this.props
+    const value = this.state.value || ''
+
+    const attributes: TextAreaAttributes = {
+      placeholder,
+      className: '',
+      name,
+      id,
+      'aria-labelledby': labelledBy,
+      value,
+      maxLength,
+      autoComplete,
+      autoCorrect,
+      title,
+      // modifiers
+      disabled,
+      readOnly,
+      required,
+      autoFocus,
+      // actions
+      onFocus,
+      onBlur,
+      onChange: this.onTextAreaChange,
+      ref: this.ref,
+    }
+
+    if (error) {
+      attributes['aria-invalid'] = 'true'
+    }
+
+    const textareaFieldClassNames = ['kirk-textarea-element']
+    const shouldDisplayButton = buttonIcon && onButtonClick && !isEmpty(buttonTitle)
+    if (shouldDisplayButton) {
+      textareaFieldClassNames.push('kirk-textarea-hasButton')
+    }
+    if (fitContent) {
+      textareaFieldClassNames.push('kirk-textarea-fitContent')
+    }
+    attributes.className = cc(textareaFieldClassNames)
+
+    return (
+      <div className={cc(['kirk-textarea', prefix({ error: !!error, disabled }), className])}>
+        {label && <label htmlFor={id}>{label}</label>}
```

Comment:
Do we want to have a label ? I don't see it in the specs.

Reply:
It was already there so I kept it.
Do we have specs to have a label for teatfields ?